torch/nn/modules/module.py

```diff
@@ -1355,25 +1380,38 @@ def register_forward_pre_hook(
         """
         handle = hooks.RemovableHandle(self._forward_pre_hooks)
         self._forward_pre_hooks[handle.id] = hook
+        if with_kwargs:
+            self._forward_pre_hooks_with_kwargs[handle.id] = True
+
         if prepend:
             self._forward_pre_hooks.move_to_end(handle.id, last=False)  # type: ignore[attr-defined]
         return handle
 
     def register_forward_hook(
-        self, hook: Callable[..., None], prepend: bool = False
+        self,
+        hook: Callable[..., None],
+        *,
+        prepend: bool = False,
+        with_kwargs: bool = False,
     ) -> RemovableHandle:
         r"""Registers a forward hook on the module.
 
         The hook will be called every time after :func:`forward` has computed an output.
         It should have the following signature::
 
-            hook(module, input, output) -> None or modified output
+            hook(module, args, output) -> None or modified output
```

Comment:
nit: you could move this signature just at the end of the with_kwargs=False paragraph below. So that each paragraph has the signature for the hook next to it.